log_collector.h

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2015 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef LOG_COLLECTOR_H
+#define LOG_COLLECTOR_H
+
+#include <string>
+#include <map>
+#include <ostream>
+
+#include "../Bricks/net/api/api.h"
+#include "../Bricks/time/chrono.h"
+#include "../Bricks/cerealize/cerealize.h"
+
+// An entry to log. Keeps everything that is easy to dig out. :-)
+struct LogEntry {
+  uint64_t t;                            // Unix epoch time in milliseconds.
+  std::string m;                         // HTTP method.
+  std::string u;                         // URL without fragments and query parameters.
+  std::map<std::string, std::string> q;  // URL query parameters.
```

Comment:
Not sure I understand the question, but two answers are:

1. Our current API for query params simply does not support multiple values per URL parameters (I know, shame), and
2. The `"u"` field, the URL, is large redundant, but I have a habit of logging everything just in case. :-)